plugins/base/base.go

```diff
@@ -33,3 +34,52 @@ type PluginInfoResponse struct {
 	// Name is the plugins name.
 	Name string
 }
+
+// NomadConfig is the nomad client configuration sent to all plugins
+type NomadConfig struct {
+	Driver *NomadDriverConfig
```

Comment:
I hate to bikeshed, but I really don't like the `Nomad` prefix on these types as it does very little to specify what this config represents.

`ClientConfig`, `AgentConfig`, or to be extremely verbose and accurate `ClientAgentConfig`

Reply:
How about `<plugin_type>Config` so `DriverConfig`?